diff --git a/dist/main.js b/dist/main.js
index 84bef16e381a6c63c3b457f22dc0dfcc8fbcab62..3312e7e34004ae29d729b4ea69f8ca5e743658d4 100644
--- a/dist/main.js
+++ b/dist/main.js
@@ -30,6 +30,9 @@ function $fbc1d750f520c94e$export$762f73dccccd255d(options) {
         ...$fbc1d750f520c94e$export$79c0c687a5963b0a({
             getKey: getKey
         }, setState),
+        setItems: (items)=>{
+            setState((prev) => ({...prev, items}))
+        },
         getItem (key) {
             return state.items.find((item)=>getKey(item) === key
             );
@@ -39,6 +42,12 @@ function $fbc1d750f520c94e$export$762f73dccccd255d(options) {
 function $fbc1d750f520c94e$export$79c0c687a5963b0a(opts, dispatch) {
     let { cursor: cursor , getKey: getKey  } = opts;
     return {
+        setItems(items) {
+            dispatch((state)=> ({
+               ...state,
+                items
+            }));
+        },
         setSelectedKeys (selectedKeys) {
             dispatch((state)=>({
                     ...state,
diff --git a/dist/module.js b/dist/module.js
index bf1d53d2a7902bd1825e1abc2ad1cd419e278620..8f60c7288466282497d6153107d225fd15f7d927 100644
--- a/dist/module.js
+++ b/dist/module.js
@@ -23,6 +23,9 @@ function $0d86e9c8f07f9a7b$export$762f73dccccd255d(options) {
         ...$0d86e9c8f07f9a7b$export$79c0c687a5963b0a({
             getKey: getKey
         }, setState),
+        setItems: (items)=>{
+            setState((prev) => ({...prev, items}))
+        },
         getItem (key) {
             return state.items.find((item)=>getKey(item) === key
             );
@@ -32,6 +35,13 @@ function $0d86e9c8f07f9a7b$export$762f73dccccd255d(options) {
 function $0d86e9c8f07f9a7b$export$79c0c687a5963b0a(opts, dispatch) {
     let { cursor: cursor , getKey: getKey  } = opts;
     return {
+        setItems(items) {
+            dispatch((state)=> ({
+                    ...state,
+                    items: items,
+                })
+            );
+        },
         setSelectedKeys (selectedKeys) {
             dispatch((state)=>({
                     ...state,
diff --git a/dist/types.d.ts b/dist/types.d.ts
index 2cf14431a5ee381e8c9024b72cfdf52d2a1c6dd2..a21c7e2809417cf7d9bc7319a4f1e19c89880d37 100644
--- a/dist/types.d.ts
+++ b/dist/types.d.ts
@@ -23,6 +23,11 @@ export interface ListData<T> {
     filterText: string;
     /** Sets the filter text. */
     setFilterText(filterText: string): void;
+    /**
+     * Set the items in the list.
+     * @param items - The items to set.
+    */
+    setItems(items: T[]): void;
     /**
      * Gets an item from the list by key.
      * @param key - The key of the item to retrieve.
