diff --git a/node_modules/@tsed/schema/lib/cjs/utils/mapNullableType.js b/node_modules/@tsed/schema/lib/cjs/utils/mapNullableType.js
index 079a10c..2b38a78 100644
--- a/node_modules/@tsed/schema/lib/cjs/utils/mapNullableType.js
+++ b/node_modules/@tsed/schema/lib/cjs/utils/mapNullableType.js
@@ -7,7 +7,7 @@ function hasNullable(obj) {
     return obj.oneOf.find((o) => o.type === "null");
 }
 function mapNullableType(obj, schema, options) {
-    if (!schema.isNullable) {
+    if (!schema?.isNullable) {
         return obj;
     }
     let types = [].concat(obj.type).filter(Boolean);
